bundles/http/Citrix/DocuSign/73f4e3b3-e2f3-43b3-9ce6-7f12da7b714c.js

```diff
@@ -0,0 +1,660 @@
+                "email_subject": envelope.emailSubject,
+                "envelope_id": envelope.envelopeId,
+                "envelope_uri": envelope.envelopeUri,
+                "expire_after": +envelope.expireAfter,
+                "expire_date_time": envelope.expireDateTime != undefined ? new Date(envelope.expireDateTime) : null,
+                "initial_sent_date_time": envelope.initialSentDateTime != undefined ? new Date(envelope.initialSentDateTime) : null,
+                "last_modified_date_time": envelope.lastModifiedDateTime != undefined ? new Date(envelope.lastModifiedDateTime) : null,
+                "purge_state": envelope.purgeState,
+                "sender_account_id": envelope.sender.accountId,
+                "sender_email": envelope.sender.email,
+                "sender_user_id": envelope.sender.userId,
+                "sender_user_name": envelope.sender.userName,
+                "sent_date_time": envelope.sentDateTime != undefined ? new Date(envelope.sentDateTime) : null,
+                "status": envelope.status,
+                "status_changed_date_time": envelope.statusChangedDateTime != undefined ? new Date(envelope.statusChangedDateTime) : null,
+                "voidedDateTime": envelope.voidedDateTime != undefined ? new Date(envelope.voidedDateTime) : null,
+                "voidedReason": envelope.voidedReason
+            }
+        })
+        dataStore.save('envelopes', templateData)
+
+        start_position += COUNT;
+        totalSetSize = json['totalSetSize'];
+        if (start_position > totalSetSize || totalSetSize == 0) {
+            i++;
+            start_position = 0
+            totalSetSize = 0
+        }
+    } while (i < userids.length)
+}
+
+//Service Actions
+async function addRecipient({ client, dataStore, integrationParameters, actionParameters }) {
+    const ACCOUNT_ID = integrationParameters.account_Id;
+    const RESEND_ENVELOPE = true
+    const respone = await client.fetch(`accounts/${ACCOUNT_ID}/envelopes/${actionParameters.envelope_Id}/recipients?resend_envelope=${RESEND_ENVELOPE}`, {
+        method: "POST",
+        body: JSON.stringify({
+            "signers": [
+                {
+                    "email": `${actionParameters.recipient_Email ?? actionParameters.recipient_Email_Unknown}`,
+                    "name": `${actionParameters.recipient_Name}`,
+                    "recipientId": `${parseInt(moment().format('HHmmss'), 10)}`,//parseInt used to remove the leading zeros
+                    "roleName": `${actionParameters.role_Name}`,
+                }
+            ]
+        })
+    })
+    const json = await respone.json()
+
+    if (!respone.ok) {
+        throw new Error(JSON.stringify(json))
+    }
+    await syncEnvelopes(client, dataStore, ACCOUNT_ID, [actionParameters.user_Id])
+}
+async function resendEnvelope({ client, dataStore, integrationParameters, actionParameters }) {
```

Comment:
This is almost the same as `addRecipient` action. I suggest merging them together with optional `roleName` parameter.

Reply:
Basically ,there was a incorrect code in resendEnvelope due to which it seems to be similar to addRecipient. Now we have done the correction in the code.